lib/iron_bank/resources/communication_profile.rb

```diff
@@ -11,6 +11,16 @@ class CommunicationProfile < Resource
       with_schema
 
       with_many :accounts
+      
+      def notifications
```

Comment:
this feels like it's too low-level -- just looking at the way the `resources/*.rb` files are done. 
maybe too support `notifications` it would be something along these lines:
```rb
with_many :notifications
```

Reply:
Sadly we can't use the `with_many` method for `notifications` on a `communication_profile`. This is because the `with_many` method uses ZOQL which is backed by SOAP API for lookup and there are no `notifications` resources in the SOAP API.

Here is the code path for `with_many` that delegates to the REST API:
- https://github.com/zendesk/iron_bank/blob/5a6aae4bd75f74a0906b6256d0824c2e7412d282/lib/iron_bank/associations.rb#L50
- https://github.com/zendesk/iron_bank/blob/5a6aae4bd75f74a0906b6256d0824c2e7412d282/lib/iron_bank/queryable.rb#L52
- https://github.com/zendesk/iron_bank/blob/5a6aae4bd75f74a0906b6256d0824c2e7412d282/lib/iron_bank/actions/query.rb#L17